### PR TITLE
TW-1293: enable recovery data in settings

### PR DESCRIPTION
**Pull request by nqhhdev**
### Ticket
- #1293 

#### Case 1: ToM => Matrix.org

https://github.com/linagora/twake-on-matrix/assets/99852347/96149eb3-a344-4c8a-b406-37b951c15375

#### Case 2: Matrix => ToM

https://github.com/linagora/twake-on-matrix/assets/99852347/fcfdf04d-ffa6-479e-b993-f1eab72e8b44

### CODE REVIEW

File: lib/widgets/matrix.dart [hunk 1]
```diff
@@ -208,6 +208,7 @@ class MatrixState extends State<Matrix>
         if (activeClient == null) return;
         final result = await setActiveClient(activeClient);
         if (result.isSuccess) {
+          setUpToMServicesInLogin(activeClient);
```
Comment (hoangdat) [inline lib/widgets/matrix.dart:211]: what is this case?

Comment (nqhhdev) [reply lib/widgets/matrix.dart:211]: First login is matrix.org not support ToM and add another account have supports ToM => handle config tom.

Before changing it we missing this case

Comment (hoangdat) [reply lib/widgets/matrix.dart:211]: IMO, all is setup inside `_registerSubs` method? what is the problem?

File: lib/pages/settings_dashboard/settings/settings.dart [hunk 2]
```diff
@@ -64,7 +68,7 @@ class SettingsController extends State<Settings> with ConnectPageMixin {
       settingEnum == optionsSelectNotifier.value;
 
   void logoutAction() async {
-    final noBackup = showChatBackupBanner == true;
+    final noBackup = showChatBackupSwitch.value == true;
```
Comment (hoangdat) [inline lib/pages/settings_dashboard/settings/settings.dart:71]: why put backup in `logout` action?

Comment (nqhhdev) [reply lib/pages/settings_dashboard/settings/settings.dart:71]: I just changed `showChatBackupSwitch` and kept old logic.
IMO: It's used for UI

File: lib/widgets/matrix.dart [hunk 3]
```diff
@@ -381,48 +370,8 @@ class MatrixState extends State<Matrix>
...
-          if (PlatformInfos.isMobile) {
-            _deletePersistActiveAccount(state);
-            TwakeApp.router.go('/home/twakeid');
-          } else {
-            TwakeApp.router.go('/home', extra: true);
-          }
-        }
-      }
-    });
+    onLoginStateChanged[name] ??= c.onLoginStateChanged.stream
```
Comment (sherlockvn) [inline lib/widgets/matrix.dart:373]: what is `c` can you rename it

**Review by sherlockvn — APPROVED** [review]

File: lib/widgets/matrix.dart [hunk 4]
```diff
@@ -194,28 +194,17 @@ class MatrixState extends State<Matrix>
...
-        if (result.isSuccess) {
-          TwakeApp.router.go(
-            '/rooms',
-            extra: LoggedInOtherAccountBodyArgs(
-              newActiveClient: activeClient,
-            ),
-          );
-          _loginClientCandidate = null;
-        }
+        Logs().d('MatrixState::getLoginClient() - Registering subs');
```
Comment (hoangdat) [inline lib/widgets/matrix.dart:206]: how about to move all of these to `_handleAddAnotherAccount()`?

**Review by hoangdat — APPROVED** [review]

**Pull request merged**